src/main/java/appPages/BaseAction.java

```diff
@@ -1,28 +1,88 @@
 package appPages;
 
 import drivers.DriverManager;
-import org.openqa.selenium.JavascriptExecutor;
-import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.WebElement;
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.FluentWait;
 import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
+import java.time.Duration;
 import java.util.Calendar;
 import java.util.Date;
+import java.util.NoSuchElementException;
+import java.util.Set;
 
 public class BaseAction {
     public WebDriver driver = DriverManager.getDriver();
+
     public void sleep(int time){
         try {
             Thread.sleep(time);
         } catch (Exception e) {
             e.printStackTrace();
         }
     }
+    public void fluentWait(By by){
+
+        FluentWait wait = new FluentWait(driver);
+        wait.withTimeout(Duration.ofSeconds(100));  // threshold is 100 seconds
+
+        wait.pollingEvery(Duration.ofSeconds(2));  // 2 | 2 | 2 |
+        wait.ignoring(NoSuchElementException.class);
+
+        wait.until(ExpectedConditions.visibilityOfElementLocated(by));
+    }
+    //sum(int a, int b) ===> sum(2,3)
```

Comment:
Code should not contain the commented line

Reply:
Done.